R/scto_read.R

```diff
@@ -19,73 +20,78 @@
 #'   any columns to datetimes.
 #' @param datetime_format String indicating format of datetimes from SurveyCTO.
 #'   See [strptime()].
+#' @param simplify Logical indicating whether to return only a `data.table`
+#'   instead of a list of `data.table`s if reading one form or dataset.
 #'
-#' @return `scto_read()` returns a `data.table`. `scto_read_all()` returns a
-#'   named list of `data.table`s, one for each form and dataset, along with a
-#'   `data.table` named ".catalog" from `scto_catalog()`.
+#' @return If `simplify` is `TRUE` and reading one form or dataset, a
+#'   `data.table`. Otherwise a named list of `data.table`s, one for each form
+#'   and dataset, along with a `data.table` named ".catalog" from
+#'   `scto_catalog()`.
 #'
 #' @examples
 #' \dontrun{
 #' auth = scto_auth('scto_auth.txt')
 #' scto_data = scto_read(auth, 'my_form')
-#' scto_db = scto_read_all(auth)
+#' scto_db = scto_read(auth)
 #' }
 #'
-#' @seealso [scto_auth()], [scto_meta()], [scto_get_attachments()],
-#'   [scto_write()]
+#' @seealso [scto_auth()], [scto_meta()], [scto_get_form_definitions()],
+#'   [scto_get_attachments()], [scto_write()]
 #'
 #' @export
 scto_read = function(
-    auth, id, start_date = '1900-01-01', review_status = 'approved',
+    auth, ids = NULL, start_date = '1900-01-01', review_status = 'approved',
```

Comment:
should default here return everything vs those approved? what is the data validation workflow? do things start off as pending and then get classified into approved or rejected?

what happens when you provide nothing for this argument to the api?

Reply:
Based on the API documentation, the default is to only return approved submissions, consistent with the behavior here.